backend/functions/matching/functions.js

```diff
@@ -86,42 +87,56 @@ async function makeMatches(courseId, groupSize = 2) {
     -(studentData.length % groupSize) || undefined //extend to the end of the array if we have perfect groups
   );
   while (i < studentDataSliced.length) {
-    let nextGroup = studentDataSliced
-      .slice(i, i + groupSize)
-      .map((studentObj) => studentObj.email);
+    let nextGroup = studentDataSliced.slice(i, i + groupSize);
     groups.push({
       groupNumber: groupCounter + lastGroupNumber,
-      members: nextGroup,
+      memberData: nextGroup, // this is to keep things consistent
     });
     i += nextGroup.length;
     groupCounter += 1;
   }
   const matchedStudentSet = new Set(studentDataSliced.map((s) => s.email));
   const unmatched = unmatchedStudents.filter((s) => !matchedStudentSet.has(s));
 
+  // map for specific firebase data (just emails, as preserved from the previous function)
+  let groupsWithMembersEmail = groups.map((group) => {
+    return {
+      groupNumber: group.groupNumber,
+      members: group.memberData.map((member) => member.email),
+    };
+  });
```

Comment:
I think what the coreball said is a good suggestion about using const instead of let just to make things more clear about what the restraints of this field is.